packages/graphql-dynamodb-transformer/src/DynamoDBModelTransformer.ts

```diff
@@ -293,10 +293,9 @@ export class DynamoDBModelTransformer extends Transformer {
             const listResolver = this.resources.makeListResolver(def.name.value, listFieldNameOverride, ctx.getQueryTypeName())
             ctx.setResource(ResolverResourceIDs.DynamoDBListResolverResourceID(typeName), listResolver)
 
-            this.generateFilterInputs(ctx, def)
-
             queryFields.push(makeModelScanField(listResolver.Properties.FieldName, def.name.value))
         }
+        this.generateFilterInputs(ctx, def)
```

Comment:
Looks good. My only feedback, as we discussed, is that this will now always generate filter inputs even when they are not used. I think this is ok, but in the future we could refactor this code such that it is called from the @connection transformer to generate filter inputs when needed.